From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <lynxplay101@gmail.com>
Date: Tue, 21 Sep 2021 15:35:07 +0200
Subject: [PATCH] Expose plugin loader addURL

To enable plugins to inject new jars into their class loader at runtime,
this commit exposes the URLClassLoader method addURL through the public
keyword.

diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 8a39c48fce819d72a94d5309db8dfc42930989af..5e545ca2f171860150667dfd3c92b96957f5653d 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -217,6 +217,13 @@ public final class PluginClassLoader extends URLClassLoader { // Spigot
         }
     }
 
+    // KTP start - expose plugin loader addURL
+    @Override
+    public void addURL(URL url) {
+        super.addURL(url);
+    }
+    // KTP end
+
     @NotNull
     Collection<Class<?>> getClasses() {
         return classes.values();
